viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -333,97 +371,81 @@ void SLAMServiceImpl::GetLatestSampledPointCloudMapString(
         }
     }
 
+    // Get data from painted surface in ARGB32 format
     auto painted_surface = painted_slices->surface.get();
+    auto image_format = cairo_image_surface_get_format(painted_surface);
+    if (image_format != cartographer::io::kCairoFormat) {
+        std::string errorLog =
+            "Error cairo surface in wrong format, expected Cairo_Format_ARGB32";
+        LOG(ERROR) << errorLog;
+        throw std::runtime_error(errorLog);
+    }
     int width = cairo_image_surface_get_width(painted_surface);
     int height = cairo_image_surface_get_height(painted_surface);
-    // Get all pixels from the painted surface in RGBA format
-    auto data = cairo_image_surface_get_data(painted_surface);
+    auto image_data = cairo_image_surface_get_data(painted_surface);
 
-    // Total number of bytes in image (4 bytes per pixel)
-    int size_data = width * height * 4;
-
-    // Each pixel contains 4 bytes of information in RGBA format
-    // data_vect[i + 0] is the R channel
-    // data_vect[i + 1] is the B channel
-    // data_vect[i + 2] is the G channel
-    // data_vect[i + 3] is the A channel
-    std::vector<unsigned char> data_vect(data, data + size_data);
+    // Get pixel containing map origin (0, 0)
+    float origin_pixel_x = painted_slices->origin.x();
+    float origin_pixel_y = painted_slices->origin.y();
 
+    // Iterate over image data and add to pointcloud buffer
     int num_points = 0;
+    std::string pcd_data;
+    for (int pixel_y = 0; pixel_y < height; pixel_y++) {
+        for (int pixel_x = 0; pixel_x < width; pixel_x++) {
+            // Get byte index associated with pixel
+            int pixel_index = pixel_x + pixel_y * width;
+            int byte_index = pixel_index * bytesPerPixel;
+
+            // We assume we are running on a little-endian system, so the ARGB
+            // order is reversed
+            pixelColorARGB color;
+            color.A = image_data[byte_index + 3];
+            color.R = image_data[byte_index + 2];
+            color.G = image_data[byte_index + 1];
+            color.B = image_data[byte_index + 0];
+
+            // Skip pixel if it contains empty data (default color)
+            if (checkIfEmptyPixel(color)) {
+                continue;
+            }
```

Comment:
[nit]
```suggestion
            ColorARGB pixel_color;
            pixel_color.A = image_data[byte_index + 3];
            pixel_color.R = image_data[byte_index + 2];
            pixel_color.G = image_data[byte_index + 1];
            pixel_color.B = image_data[byte_index + 0];

            // Skip pixel if it contains empty data (default color)
            if (CheckIfEmptyPixel(pixel_color)) {
                continue;
            }
```

Reply:
moving all variables non consts to snake case due to google C++ style guidelines